### PR TITLE
[CNDIT-1293] Post processing investigation dimension logic

### PR DESCRIPTION
**Pull request by rshanmugam-enquizit**
### Notes
This PR includes the logic for the the call to `sp_nrt_investigation_postprocessing` to update the `d_investigation`, `d_investigation_confirmation_method`, `d_investigation_notification` and `d_investigation_observation` dimension tables in the post processing service.

### JIRA
- https://cdc-nbs.atlassian.net/browse/CNDIT-1293

### Testing
- Unit tests with 90% coverage added (except model classes).

### CODE REVIEW

**Review by upasanapattnaik-eq — APPROVED** [review]
LGTM. Thank you!

File: data-reporting-service/post-processing-service/src/main/resources/application.yaml [hunk 1]
```diff
@@ -2,8 +2,6 @@ spring:
   kafka:
     topic:
       investigation: nrt_investigation
-      investigation-confirmation: nrt_investigation_confirmation
-      investigation-notification: nrt_investigation_notification
       observation: nrt_observation
       organization: nrt_organization
       patient: nrt_patient
```
Comment (alog-enquizit) [inline data-reporting-service/post-processing-service/src/main/resources/application.yaml:7]: Consider Updating the port number from 8090 to something different. We are using the same port for the other services too.  Prod deployment won't be an issue but locally running multiple services on the same port might throw some issues.

File: data-reporting-service/post-processing-service/src/test/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingServiceTest.java [hunk 2]
```diff
@@ -59,7 +61,7 @@ public void testPostProcessPatientMessage() {
         postProcessingServiceMock.postProcessPatientMessage(key, topic);
         postProcessingServiceMock.processCachedIds();
 
-        String expectedPatientIdsString = "123"; //String.join(",", patientIds.stream().map(String::valueOf).collect(Collectors.toList()));
+        String expectedPatientIdsString = "123";
         verify(patientRepositoryMock).executeStoredProcForPatientIds(expectedPatientIdsString);
         verify(patientRepositoryMock, times(1)).executeStoredProcForPatientIds("123");
```
Comment (alog-enquizit) [inline data-reporting-service/post-processing-service/src/test/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingServiceTest.java:66]: you can use `expectedPatientIdsString` here instea of hardcoding.

**Review by alog-enquizit — APPROVED** [review]
lgtm 👍

File: data-reporting-service/post-processing-service/src/main/resources/application.yaml (hunk 1 above)
Comment (sveselev) [reply data-reporting-service/post-processing-service/src/main/resources/application.yaml:7]: > Consider Updating the port number from 8090 to something different. We are using the same port for the other services too. Prod deployment won't be an issue but locally running multiple services on the same port might throw some issues.

Just FYI, the port 8093 is taken for investigation preprocessing service (the PR under review)

File: data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java [hunk 3]
```diff
@@ -98,7 +82,7 @@ protected void processCachedIds() {
                 if(keyTopic.contains("investigation")) {
                     String idsString = String.join(",", ids.stream().map(String::valueOf).collect(Collectors.toList()));
                     logger.info("Processing the ids from the topic {} and calling the stored proc for patient: {}", keyTopic, idsString);
```
Comment (sveselev) [inline data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java:84]: All the logger messages here refer to 'patient'

**Review by sveselev — APPROVED** [review]

File: data-reporting-service/post-processing-service/src/main/resources/application.yaml (hunk 1 above)
Comment (rshanmugam-enquizit) [reply data-reporting-service/post-processing-service/src/main/resources/application.yaml:7]: when running in local, this is always going to be an issue. so, let's leave it this way and user can fix when running it in local

File: data-reporting-service/post-processing-service/src/test/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingServiceTest.java (hunk 2 above)
Comment (rshanmugam-enquizit) [reply data-reporting-service/post-processing-service/src/test/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingServiceTest.java:66]: will fix it

**Pull request merged**